File: Backend/football/src/main/java/com/example/backend/entity/Event.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Getter
@Setter
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@ToString
@EqualsAndHashCode
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idEvent;

    private Integer minute;

    @Enumerated(EnumType.STRING)
    private Type type;

    @ManyToOne
    @JoinColumn(name = "player_id")
    private Player player;

    @ManyToOne
    @JoinColumn(name = "match_id")
    private Match match;

    @ManyToOne
    @JoinColumn(name = "assist_player_id", nullable = true)
    private Player assist;

    private Boolean onTarget;

    public enum Type {
        GOAL, ASSIST, SHOT, TACKLE, PASS
    }
}

File: Frontend/matchanalytics/src/app/models/Event..ts
import { Player } from '../models/Player.';

export type EventType = 'GOAL'|'ASSIST'|'SHOT'|'TACKLE'|'PASS';

export interface MatchEvent {
  idEvent?: number;
  minute: number;
  type: EventType;
  player: Player;
  match?: { idMatch: number };
  assist?: Player | null;
  onTarget?: boolean;
}
